src/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/StartupCode/StartupCodeHelpers.cs

```diff
@@ -18,6 +20,7 @@ internal static class StartupCodeHelpers
         internal static void Initialize()
         {
             InitializeStringTable();
+            RunEagerClassConstructors();
             RuntimeImports.RhEnableShutdownFinalization(0xffffffffu);
```

Comment:
We will be able to get rid of this special code and switch to the Project N plan (this will be triggered from the eager cctor of the Environment class).